services/sso/src/main/java/software/amazon/awssdk/services/sso/internal/SsoAccessToken.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sso.internal;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.auth.token.credentials.SdkToken;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+
+@SdkInternalApi
+public class SsoAccessToken implements SdkToken {
+    private final String accessToken;
+    private final Instant expiresAt;
+
+    private SsoAccessToken(BuilderImpl builder) {
+        Validate.paramNotNull(builder.accessToken, "accessToken");
+        Validate.paramNotNull(builder.expiresAt, "expiresAt");
+        this.accessToken = builder.accessToken;
```

Comment:
Nit: can be shortened to `this.accessToken = Validate.paramNotNull(builder.accessToken, "accessToken");`